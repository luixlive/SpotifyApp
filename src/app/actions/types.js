// Device reducer
export const DEVICE_TYPE_CHANGED = 'DEVICE_TYPE_CHANGED';

// User
export const LOAD_USER = 'LOAD_USER';
export const LOAD_USER_FAILED = 'LOAD_USER_FAILED';
export const LOAD_USER_SUCCEEDED = 'LOAD_USER_SUCCEEDED';
export const LOGOUT_USER = 'LOGOUT_USER';
export const LOGOUT_USER_FAILED = 'LOGOUT_USER_FAILED';
export const LOGOUT_USER_SUCCEEDED = 'LOGOUT_USER_SUCCEEDED';

// UserStats
export const LOAD_USER_STATS = 'LOAD_USER_STATS';
export const LOAD_USER_STATS_FAILED = 'LOAD_USER_STATS_FAILED';
export const LOAD_USER_STATS_SUCCEEDED = 'LOAD_USER_STATS_SUCCEEDED';
