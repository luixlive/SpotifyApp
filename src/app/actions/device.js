import { DEVICE_TYPE_CHANGED } from './types';

export default deviceMobile => (
  { type: DEVICE_TYPE_CHANGED, payload: { deviceMobile } }
);
