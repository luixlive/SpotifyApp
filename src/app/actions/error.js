import { CLEAR_ERROR } from './types';

export default () => (
  { type: CLEAR_ERROR, payload: {} }
);
