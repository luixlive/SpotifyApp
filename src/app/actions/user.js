import { LOAD_USER, LOGOUT_USER } from './types';

export const loadUser = () => (
  { type: LOAD_USER, payload: { } }
);

export const logoutUser = () => (
  { type: LOGOUT_USER, payload: { } }
);
