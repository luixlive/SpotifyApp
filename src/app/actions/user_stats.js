import { LOAD_USER_STATS } from './types';

export default () => (
  { type: LOAD_USER_STATS, payload: { } }
);
