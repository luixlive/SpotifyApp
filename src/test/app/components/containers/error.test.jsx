import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';

import Error, {
  PureError,
} from './../../../../app/components/containers/error';
import initialState from './../../../test_utils/initial_state';
import injectRouter from './../../../test_utils/inject_router';
import * as types from './../../../../app/actions/types';
import {
  error as componentProps,
} from './../../../test_utils/components_props';

describe('App Components - Error', () => {
  const mockStore = configureStore();
  let props;
  let store;
  beforeEach(() => {
    props = _.cloneDeep(componentProps);
    store = mockStore(initialState);
  });

  describe('Snapshots', () => {
    it('renders', () => {
      const rendered = renderer.create(injectRouter(() => (
        <Provider store={store}>
          <PureError {...props} />
        </Provider>
      ))).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {
    it('calls clearError and loadUser when user clicks link', () => {
      const clearError = jest.fn();
      const loadUser = jest.fn();
      const wrapper = mount(injectRouter(() => (
        <Provider store={store}>
          <PureError
            {...props}
            clearError={clearError}
            loadUser={loadUser}
          />
        </Provider>
      )));
      wrapper.find('Link').simulate('click');
      expect(loadUser).toHaveBeenCalledTimes(1);
      expect(clearError).toHaveBeenCalledTimes(1);
    });
  });

  describe('Provider', () => {
    let wrapper;
    beforeEach(() => {
      store = mockStore(initialState);
      wrapper = mount(injectRouter(() => (
        <Provider store={store}>
          <Error />
        </Provider>
      )));
    });

    it('renders', () => {
      expect(wrapper.find(PureError).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(Object.keys(wrapper.find(PureError).props()))
        .toEqual(Object.keys(componentProps));
    });

    it('dispatches clearError and loadUser', () => {
      expect(store.getActions().length).toBe(0);
      wrapper.find('Link').simulate('click');

      const expectedActions = [{
        type: types.CLEAR_ERROR,
        payload: {},
      }, {
        type: types.LOAD_USER,
        payload: {},
      }];
      expect(store.getActions()[0]).toEqual(expectedActions[0]);
      expect(store.getActions()[1]).toEqual(expectedActions[1]);
    });
  });
});
