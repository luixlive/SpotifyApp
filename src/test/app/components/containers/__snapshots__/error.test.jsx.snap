// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - Error Snapshots renders 1`] = `
<div
  className="ui center aligned container"
>
  <h1
    className="ui header"
    style={
      Object {
        "fontSize": "4em",
        "marginTop": "1.5em",
      }
    }
  >
    Error :(
  </h1>
  <h2
    className="ui header"
  >
    
     Go back to
    <a
      href="/"
      onClick={[Function]}
    >
       Home
    </a>
    .
  </h2>
</div>
`;
